### PR TITLE
Add whitelisted holders

### PR DESCRIPTION
**Pull request by surangap**
### Changelog
- Add whitelisted holders feature to doughnuts

### CODE REVIEW

File: pallet/doughnut/src/lib.rs [hunk 1]
```diff
@@ -313,6 +318,8 @@ pub mod pallet {
 		TRNNutPermissionDenied,
 		/// Inner call is not whitelisted
 		UnsupportedInnerCall,
+		/// Not a whitelisted holder
+		NotAWhitelistedHolder,
```
Comment (JasonTulp) [inline pallet/doughnut/src/lib.rs:322]: nit
```suggestion
		HolderNotWhitelisted,
```
or "HolderNotAllowed"

**Review by JasonTulp — APPROVED** [review]
lgtm!

File: pallet/doughnut/src/lib.rs [hunk 2]
```diff
@@ -419,6 +432,19 @@ pub mod pallet {
...
+		// Note: this is for temporary purpose. Might change in the future
+		#[pallet::weight(T::WeightInfo::update_whitelisted_holders())]
+		pub fn update_whitelisted_holders(
+			origin: OriginFor<T>,
+			holder: T::AccountId,
+			add: bool,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+			WhitelistedHolders::<T>::set(holder, add);
+			Ok(())
```
Comment (zees-dev) [inline pallet/doughnut/src/lib.rs:446]: Need to always emit event when changing pallet state

**Review by zees-dev — COMMENTED** [review]
Need to emit event for the extrinsic; otherwise LGTM 👍

**Pull request merged**